siliconcompiler/tools/yosys/syn_asic.tcl

```diff
@@ -2,40 +2,32 @@
 # DESIGNER's CHOICE
 ####################
 
-set sc_targetlibs  [dict get $sc_cfg asic logiclib]
-set sc_macrolibs   [dict get $sc_cfg asic macrolib]
-set sc_mainlib     [lindex [dict get $sc_cfg asic logiclib] 0]
-set sc_delaymodel  [dict get $sc_cfg asic delaymodel]
-set sc_tie         [dict get $sc_cfg library $sc_mainlib asic cells tie]
-set sc_buf         [dict get $sc_cfg library $sc_mainlib asic cells buf]
-set sc_scenarios   [dict keys [dict get $sc_cfg constraint]]
+set sc_libraries        [dict get $sc_cfg tool $sc_tool var $sc_step $sc_index synthesis_libraries]
+if {[dict exists $sc_cfg tool $sc_tool var $sc_step $sc_index synthesis_libraries_macros]} {
+    set sc_macro_libraries [dict get $sc_cfg tool $sc_tool var $sc_step $sc_index synthesis_libraries_macros]
+} else {
+    set sc_macro_libraries []
+}
+set sc_mainlib          [lindex [dict get $sc_cfg asic logiclib] 0]
+set sc_tie              [dict get $sc_cfg library $sc_mainlib asic cells tie]
+set sc_buf              [dict get $sc_cfg library $sc_mainlib asic cells buf]
 
 ########################################################
 # Read Libraries
 ########################################################
 
-set stat_libs ""
-
-foreach item $sc_scenarios {
-    foreach libcorner [dict get $sc_cfg constraint $item libcorner] {
-	foreach lib $sc_targetlibs {
-	    if {[dict exists $sc_cfg library $lib model timing $sc_delaymodel $libcorner]} {
-		puts "SC Reading liberty file (corner=$libcorner, lib=$lib, mode=$sc_delaymodel)"
-		set lib_file [dict get $sc_cfg library $lib model timing $sc_delaymodel $libcorner]
-		yosys read_liberty -lib $lib_file
-	    }
-	}
-	# Yosys doesn't handle macro PG pins properly, so we don't use the .libs here
-	# Instead we use black boxes for all macros.
-	foreach lib $sc_macrolibs {
-	    if {[dict exists $sc_cfg library $lib model timing $sc_delaymodel $libcorner]} {
-		puts "SC: Reading liberty file (corner=$libcorner, lib=$lib, mode=$sc_delaymodel)"
-		set lib_file [dict get $sc_cfg library $lib model timing $sc_delaymodel $libcorner]
-		yosys read_liberty -lib $lib_file
-		append stat_libs "-liberty $lib_file "
-	    }
-	}
-    }
+set lib_args ""
```

Comment:
Are there opportunities in the schema/python to put more on the emphasis on the python wrapper for the tool to simplfy the TCL code? Having logic that straddles tcl/python bondary has always beeb a proble, epsecially when the TCL interface is limited like in Yosys.

Reply:
Are you suggesting that the python wrapper generate a list for the stat and abc commands and pass that in instead? So we don't build the argument list here? That would certainly simplify some of the openroad scripts

Reply:
I changed it, such that the path is set by the python script, however, it is still a constant (just now it's controlled in one place and from the python tool driver)

Reply:
I moved this code around a little to simplify it a bit in each place